scripts/wsrep_sst_clone.sh

```diff
@@ -0,0 +1,1251 @@
+#              │Joiner get IST    │
+#              │or SST.If IST     │
+#              │Bypass all and    │
+#              │Wait for IST.     │
+#              │SST Start clone   │
+#              │Instance and wait │
+#              │for donor to clone│
+#              └─────────┬────────┘
+#              ┌─────────▼─────────────┐
+#              │Clone process is       │
+#              │reported.              │
+#              │When done Dn waits     │
+#              │Joiner close instance  │
+#              │And performs 2 restarts│
+#              └──────────┬────────────┘
+#              ┌──────────▼──────────────┐
+#              │1) To fix dictionary  &  │
+#              │   Final cleanup         │
+#              │2) To recover position   │
+#              └──────────┬──────────────┘
+#              ┌──────────▼──────────────┐
+#              │Send final ready signal  │
+#              │Waits for IST            │
+#              └─────────────────────────┘
+#----------------------------------------------------------------------------
+# Config variables:
+# [sst]
+# netcat_port=4442
+# wsrep-debug=true
+#
+# NOTE for SSL usage
+# When using clone SSL certificates MUST be manually (or by automation) placed in a location on both servers DONOR/RECEIVER
+# this location better not be the data directory given it is manipulated by the clone process.
+# It is reccomended to explicitly set the ssl certificates in the my.cnf as:
+# [client]
+#   ssl-ca = /<path>/ca.pem
+#   ssl-cert = /<path>/client-cert.pem
+#   ssl-key = /<path>/client-key.pem
+# [mysqld] 
+#   ssl-ca = /<path>/ca.pem
+#   ssl-cert = /<path>/server-cert.pem
+#   ssl-key = /<path>/server-key.pem
+# OR
+#  [mysqld]
+#   clone_ssl_ca=/path/to/ca.pem
+#   clone_ssl_cert=/path/to/client-cert.pem
+#   clone_ssl_key=/path/to/client-key.pem   
+# Where <path> is not the data directory  
+#############################################################################
+
+CMDLINE="$0 $*"
+
+set -o nounset -o errexit
+
+readonly EINVAL=22
+readonly EPIPE=32
+readonly ETIMEDOUT=110
+CLEANUP_CLONE_PLUGIN=""
+CLEANUP_CLONE_SSL=""
+CLONE_USER=""
+PARENT_PID=""
+NC_PID=""
+CLONE_INSTANCE_PID=""
+
+# The following variable will store the position (GTID:POS) coming from the donor in the message exchange. 
+# It will be used ONLY in emergency as last measure if for any reason the position recovery (at the end) will fail 
+RP_PURGED_EMERGENCY=""
+WSREP_SST_OPT_LPORT=""
+WSREP_SST_OPT_PARENT=""
+#Default to 4444 as we do for XB
+CLONE_INSTANCE_PORT=4444
+
+
+# wsrep_gen_secret() generates enough randomness, yet some MySQL password
+# policy may insist on having upper and lower case letters, numbers and
+# special symbols. Make sure that whatever we generate, it has it.
+readonly PSWD_POLICY="Aa"
+
+OS=$(uname)
+[ "$OS" = "Darwin" ] && export -n LD_LIBRARY_PATH
+
+. $(dirname "$0")/wsrep_sst_common #_clone
+
+JOINER_TIMEOUT_WAIT_XST=$(parse_cnf sst joiner-timeout-xst "60")
```

Comment:
What does XST mean? Can we rename it to something that is more readable?

Reply:
Renamed to JOINER_TIMEOUT_WAIT_DONOR_MESSAGE and joiner-timeout-wait-donor-message